plugin/storage/es/factory.go

```diff
@@ -317,9 +310,35 @@ func (f *Factory) CreateSamplingStore(maxBuckets int) (samplingstore.Store, erro
 		Lookback:               f.primaryConfig.AdaptiveSamplingLookback,
 		MaxDocCount:            f.primaryConfig.MaxDocCount,
 	})
+
+	if f.primaryConfig.CreateIndexTemplates && !f.primaryConfig.UseILM {
+		mappingBuilder := getMappingBuilder(f.primaryConfig, f.templateBuilder)
+		sampleMapping, err := mappingBuilder.GetSamplingMappings()
+		if err != nil {
+			return nil, err
+		}
+		if err := store.CreateTemplates(sampleMapping); err != nil {
+			return nil, err
+		}
+	}
+
 	return store, nil
 }
 
+func getMappingBuilder(cfg *config.Configuration, templateBuilder es.TemplateBuilder) mappings.MappingBuilder {
+	return mappings.MappingBuilder{
+		TemplateBuilder:              templateBuilder,
```

Comment:
why do you need this to be passed as parameter? Previously it was initialized in-place:
```
	mappingBuilder := mappings.MappingBuilder{
		TemplateBuilder:              es.TextTemplateBuilder{},
```

Reply:
I reverted the code back to its original state. eaa6bdce979d6f4779ffe52df358a43fcf5ac38a